plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/perspective/widgets/machine/panel/MachineTreeNode.java

```diff
@@ -48,8 +47,8 @@ public MachineTreeNode(@Assisted MachineTreeNode parent,
 
         boolean isMachine = data instanceof Machine;
 
-        id = isMachine ? ((MachineDto)data).getId() : ROOT;
-        name = isMachine ? ((MachineDto)data).getConfig().getName() : ROOT;
+        id = isMachine ? ((Machine)data).getId() : ROOT;
```

Comment:
Hm, I meant using if/else condition instead of ternary operator something like 
```java
if (data instanceof MachineDto){
   MachineDto machineDto = (MachineDto)data;
   MachineConfig config = machineDto.getConfig();
   name = config.getName();
} else {
   Log.warning(getClass(), "Data type is not a machine setting default value");
   name = ROOT;
}
```
So we could perform decomposition of complex actions, conditional type casting along with additional (like logging) operations based on some conditions. That is just a proposal, so first of all I'm interested in your opinion